includes/CLI/Plugin_Check_Command.php

```diff
@@ -266,35 +424,94 @@ private function get_options( $assoc_args ) {
 		return $options;
 	}
 
+	/**
+	 * Validates the associative arguments 'list-check-categories' command.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if format is invalid.
+	 */
+	private function get_list_check_categories_options( $assoc_args ) {
+		$defaults = array(
+			'format' => 'table',
+		);
+
+		$options = wp_parse_args( $assoc_args, $defaults );
+
+		if ( ! in_array( $options['format'], $this->output_formats, true ) ) {
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Output formats.
+					__( 'Invalid format argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->output_formats )
+				)
+			);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Validates the associative arguments 'list-checks' command.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if format is invalid.
+	 */
+	private function get_list_checks_options( $assoc_args ) {
+		$defaults = array(
+			'format' => 'table',
+		);
+
+		$options = wp_parse_args( $assoc_args, $defaults );
+
+		if ( ! in_array( $options['format'], $this->output_formats, true ) ) {
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Output formats.
+					__( 'Invalid format argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->output_formats )
+				)
+			);
+		}
+
+		return $options;
+	}
+
 	/**
 	 * Gets the formatter instance to format check results.
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param array $assoc_args Associative arguments.
+	 * @param array  $assoc_args Associative arguments.
+	 * @param string $type       Command type.
 	 * @return WP_CLI\Formatter The formatter instance.
 	 */
-	private function get_formatter( $assoc_args ) {
-		$default_fields = array(
-			'line',
-			'column',
-			'code',
-			'message',
-		);
+	private function get_formatter( $assoc_args, $type ) {
+		$default_fields = $this->default_fields[ $type ];
```

Comment:
Why do we need a new class property for this? This method is only called once in each check, so I think it's simpler and more intuitive and efficient to pass the actual default fields to this method rather than passing the type and then looking up the default fields.

Reply:
Private property removed.